_posts/2023-01-18-tuf.md

```diff
@@ -0,0 +1,661 @@
+artifacts, and would specify the metadata and targets to download:
+```console
+$ tuftool download \
+   --root root/root.json \
+   --targets-url "file://$PWD/repo/targets" \
+   --metadata-url "file://$PWD/repo/metadata" \
+   downloaded-repo
+```
+And we can inspect the downloaded file:
+```console
+$ cat downloaded-repo/artifact_1.txt 
+version 1 of artifact_1
+```
+
+Alright, now lets say we need create a new release of our target,
+artifact_1.txt. To do this we would use the tuftool update command. First we
+need to make a change to our artifact:
+```console
+$ echo "version 2 of artifact_1" > artifacts/artifact_1.txt
+```
+Now we can run the update command:
+```console
+$ tuftool update \
+   --root root/root.json \
+   --key keys/root.pem \
+   --add-targets artifacts \
+   --targets-expires 'in 3 weeks' \
+   --targets-version 2 \
+   --snapshot-expires 'in 3 weeks' \
+   --snapshot-version 2 \
+   --timestamp-expires 'in 1 week' \
+   --timestamp-version 2 \
+   --outdir repo \
+   --metadata-url file:///$PWD/repo/metadata
+```
+After that command has been run there will have been a changes to
+`repo/metadata/timestamp.json`:
+```console
+$ cat repo/metadata/timestamp.json 
+{
+  "signed": {
+    "_type": "timestamp",
+    "spec_version": "1.0.0",
+    "version": 2,
+    "expires": "2023-01-24T12:49:36.580297365Z",
+    "meta": {
+      "snapshot.json": {
+        "length": 1004,
+        "hashes": {
+          "sha256": "05e44a53c09c018f812a7ffb59ecd147d4265c6ca2b10ba4e56a46397fb5d7ff"
+        },
+        "version": 2
+      }
+    }
+  },
+  "signatures": [
+    {
+      "keyid": "6e99ec437323f2c7334c8b16fd7a7a197829ba89ff50d07aa4b50fc9634dad9f",
+      "sig": "73995d97b2df1ebe2044152d054febad55100f0699ec199ebd1e237c57242c9e536a5a53b9198eda8f965c1434c9b75829b7a34f6294176e9cee24b140a0737fb3c1c3b7fe13466c4ea2622b18177e78a9440cfd2e66fd46f2041bcb139722e175491a355f83f9c8f7f8d44366daddadb79a123d9d4d8886eb46f92a8e16a2900beb3daeba72d6fa815b82d7ab49af4246e2148dbd41854eef309aabb7ca6598ca52195357b4d0537bdd61a228cce2c8ee24d66a0465b528a2552a9fac65497619480c62b2beecfea67f05994eb39bf021800eeb0fede3f4dcf1e8c1211cf429a0e8db2fcd5c2680d4873911eeb1e4067b9d7b50aa60e8668cfbb01ba343d901"
+    }
+  ]
+}
+```
+And we have two new metadata files, `repo/metadata/2.snapshot.json`, and
+`repoo/metadata/2.targets.json`.
+
+So lets simulate our downloading from the tuf server again:
+```console
+$ tuftool download \
+   --root root/root.json \
+   --targets-url "file://$PWD/repo/targets" \
+   --metadata-url "file://$PWD/repo/metadata" \
+   downloaded-repo2
+```
+And lets take a look at the artifact:
+```console
+$ cat downloaded-repo2/artifact_1.txt 
+version 2 of artifact_1
+```
+
+### TUF usage in Sigstore
+Sigstore uses TUF to protect their keys and is one reason for trying to say
+things instead of software updates in the text above.
+
+__TODO__: expand on this...or is it too much for a single post?
```

Comment:
I would really love to see this content, but it could be a post of its own I guess.

Reply:
I think it might make sense to show/explain how Sigstore uses TUF for its keys, showing what their TUF repo looks like which should hopefully look familiar after reading the first section of this post. 

And then perhaps a new post about TUF delegation and then also show how Sigstore uses (or will use, not sure about the state of things yet) that, but I need to do some investigation/learning around that first.